src/Microsoft.Extensions.HealthChecks.Abstractions/HealthCheckResult.cs

```diff
@@ -0,0 +1,182 @@
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.HealthChecks
+{
+    /// <summary>
+    /// Represents the result of a health check.
+    /// </summary>
+    public struct HealthCheckResult
+    {
+        private static readonly IReadOnlyDictionary<string, object> _emptyReadOnlyDictionary = new Dictionary<string, object>();
+
+        private string _description;
+        private IReadOnlyDictionary<string, object> _data;
+
+        // REVIEW: Put the name of the health check or IHealthCheck instance on here?
```

Comment:
Either that of the dictionary-based approach. I'm not sure you need both. Leaving it off is easier for implementors

Reply:
Yeah, and the dictionary is useful for consumers of `IHealthCheckService` who want to run specific checks.